commands/cli/parse.go

```diff
@@ -75,67 +58,142 @@ func Parse(input []string, stdin *os.File, root *cmds.Command) (cmds.Request, *c
 	return req, cmd, path, nil
 }
 
-// parsePath separates the command path and the opts and args from a command string
-// returns command path slice, rest slice, and the corresponding *cmd.Command
-func parsePath(input []string, root *cmds.Command) ([]string, []string, *cmds.Command) {
-	cmd := root
-	path := make([]string, 0, len(input))
-	input2 := make([]string, 0, len(input))
+// Parse a command line made up of sub-commands, short arguments, long arguments and positional arguments
+func parseOpts(args []string, root *cmds.Command) (
+	path []string,
```

Comment:
soooo many return values, I cant help but wonder if theres a better way to structure this all in a future PR